### ISSUE TITLE
The server doesn't recompile the files on change

### ISSUE THREAD

**Issue by 1234filip** [webpack/webpack-dev-server#2856]
- Operating System: Windows 10 x64(host), WSL2: Ubuntu 20.04
- Node Version: v12.118.2
- NPM Version: 6.14.5
- webpack Version: 5.6.0
- webpack-dev-server Version: 3.11.0
- Browser: Chrome

- [x] This is a **bug**
- [ ] This is a **modification** request

### Code

```js
/* eslint-disable no-undef */
// Webpack uses this to work with directories
const CopyWebpackPlugin = require("copy-webpack-plugin");
const MiniCssExtractPlugin = require("mini-css-extract-plugin");
const path = require("path");

// This is the main configuration object.
// Here you write different options and tell Webpack what to do
module.exports = {
    // Path to your entry point. From this file Webpack will begin his work
    entry: {
        babel: "babel-polyfill",
        index: "./static/src/index.js",
        base: {
            import: "./static/src/js/base.js",
            dependOn: "shared",
        },
        testDetail: {
            import: "./static/src/js/test-detail.js",
            dependOn: "shared",
        },
        upload: {
            import: "./static/src/js/upload.js",
            dependOn: "shared",
        },
        shared: "cash-dom",
    },

    // Path and filename of your result bundle.
    // Webpack will bundle all JavaScript into this file
    output: {
        path: path.resolve(__dirname, "static/dist"),
        publicPath: "/static/",
        filename: "js/[name].bundle.js",
    },
    devtool: "inline-source-map",
    module: {
        rules: [
            {
                test: /\.js$/,
                exclude: /(node_modules)/,
                use: {
                    loader: "babel-loader",
                    options: {
                        presets: ["@babel/preset-env"],
                    },
                },
            },
            {
                test: /\.(sa|sc|c)ss$/,
                use: [MiniCssExtractPlugin.loader, "css-loader", "sass-loader"],
            },
        ],
    },
    plugins: [
        new MiniCssExtractPlugin({
            filename: "css/app.css",
        }),
        new CopyWebpackPlugin({
            patterns: [{ from: "static/src/images", to: "images" }],
        }),
    ],
    watchOptions: {
        poll: true,
        aggregateTimeout: 300,
    },
    devServer: {
        contentBase: "./static/dist/",
        publicPath: "/static/",
        writeToDisk: true,
        proxy: {
            "!/static/**": {
                target: "http://localhost:8000", // points to django dev server
                changeOrigin: true,
            },
        },
    },
    optimization: {
        runtimeChunk: "single",
    },
};

```

### Expected Behavior
Webpackdevserver server reloads when I change the code

### Actual Behavior
When changing the files the server doesn't recompile them or refresh the page. I'm using `--hot` option too.

### For Bugs; How can we reproduce the behavior?
1. Run `yarn webpack serve --hot --env development --mode development`
2. Wait for the server to start and try to change a file that is an entry point. Doesn't do anything.

Disclaimer: It seems to be a bug with the newer version of webpack or with the `webpack server` command. On the older versions of webpack where I didn't get the #2759 issue, the reloading worked fine with `yarn webpack-dev-server`.

**Comment by 1234filip** [comment]
Note: seems to work with `webpack-cli@3.3.12` and `webpack-dev-server` instead of `webpack serve`

**Comment by evilebottnawi** [comment]
Please create reproducible test repo, I think it is fixed for v4

**Comment by alexander-akait** [comment]
Please try https://github.com/webpack/webpack-dev-server/releases/tag/v4.0.0-beta.0, feel free to feedback if it is not solve the problem

**Comment by Semigradsky** [comment]
👍 recompiling fixed for me in v4

**Comment by alexander-akait** [comment]
Workaround for v3 - `target: 'web'`, fixed in `v4.0.0.-beta.0`
